### PR TITLE
K8 Unit tests

### PR DESCRIPTION
**Pull request by Philip-21**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 
#### All unit tests for  k8 packages in Meshkit
K8 unit tests for describe package continuation from pr #306 and #287

### CODE REVIEW

**Comment by Aisuko** [comment]
> @Revolyssup pls I will need the Pr to be reviewed, I feel I am really missing something in the code to make the test case work . As you had said earlier, i created and initialized a test.yml file to be used as my meshclient. When running the testcase it gave this error message , “A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.”
> 
> @Aisuko mentioned using a controller-runtime to create a response. I'll need your review on this to continue working with the Pr

Yes, we have done this in `meshery-operator` as you mentioned above. For `meshery/mshkit`. It would be better if we can get some suggestions from @Revolyssup. As I mentioned before, please get a confirmation of the solution, after that it is good to go.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Please check the CI issues

**Comment by Philip-21** [comment]
> Please check the CI issues

i have checked the ci issues it has to do with the response errors i mentioned  earlier, i have pinged @Revolyssup  i am still waiting for his reply and suggestions

File: utils/kubernetes/describe/describe_test.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	clienttime := &http.Client{
+		Timeout:   timeout,
+		Transport: http.DefaultTransport,
+	}
+	httpmock.ActivateNonDefault(clienttime)
+	//set up mock client  to handle api request and return expected responses
+	mckpod := MockClient{
+		Object:       "Pod",
+		Method:       "GET",
+		RequestUrl:   fmt.Sprintf("%sapi/v1/namespaces", ctx),
```
Comment (Revolyssup) [inline utils/kubernetes/describe/describe_test.go:37]: @Philip-21 You dont need to use `ctx` here as the prefix. Use the address given in test.yml

**Review by Revolyssup — CHANGES_REQUESTED** [review]

**Comment by Aisuko** [comment]
Hi, @Philip-21 Are you still working on it?

**Comment by Philip-21** [comment]
> Hi, @Philip-21 Are you still working on it?

Yes, i will create time and look into this

**Pull request closed without merging**